src/main/java/javapoz24/team3/rental/application/config/MockData.java

```diff
@@ -0,0 +1,50 @@
+package javapoz24.team3.rental.application.config;
+
+import javapoz24.team3.rental.application.RentalService;
+import javapoz24.team3.rental.domain.rental.Address;
+import javapoz24.team3.rental.domain.rental.CompanyBranch;
+import javapoz24.team3.rental.domain.rental.Rental;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.PostConstruct;
+import java.net.URI;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+@Service
+public class MockData {
+
+    @Autowired
+    private RentalService rentalService;
+
+    @PostConstruct
+    public void GenerateMockData() {
+
+        Address addressOfBranch1 = new Address("Dąbrowskiego 345", "Poznań", "61-234", null);
+        Address addressOfBranch2 = new Address("Trzebnicka 12", "Wrocław", "50-015", null);
+        Address addressOfBranch3 = new Address("Domaniewska 44", "Warszawa", "00-122", null);
+        CompanyBranch branch1 = new CompanyBranch(addressOfBranch1, null);
+        CompanyBranch branch2 = new CompanyBranch(addressOfBranch2, null);
+        CompanyBranch branch3 = new CompanyBranch(addressOfBranch3, null);
+
```

Comment:
gdzieś tutuaj powinno być łączenie adresu z CompanyBranch.
np
CompanyBranch branch3 = new CompanyBranch(addressOfBranch3, addressOfBranch3);

Reply:
Hibernate powinien sam zapisać address jeżeli zapiszesz CompanyBranch. Jeżli to się nie stanie to trzeba dodać @OneToOne(cascade= CascadeType.PERSIST) albo inna flagę zależy co chcesz osiągnąć.

Reply:
Adres na razie jest już osadzony (Embedded). Problem mam z tym, że nie zapisuje Oddziałów do bazy gdy zapisuje cały Rental. CascadeType ALL i PERSIST nic nie zmienia. Po odczycie z bazy nie ma nic w kolekcji companyBranches. 
Czy trzeba jest zapisywać osobno do bazy ze wskazaniem powiązania do Rental? - Czyli najpierw tworzę Rental bez odziałów, zapisuję, potem zapisuję oddziały osobno.
Jak przypiszę do obiektów companyBranches pole Rental (wkażę obiekt rental a nie null)i próbuję tak zapisać cały Rental (z kolekcją companyBranches) to wyrzuca StackOverflow.

Reply:
Musiałbym zobaczyć stacktrace tego SO ale pewnie problem jest taki że poprzez lomboka przy przypisaniu do seta na equals wywoluje sie w nieskonczoność

Reply:
Tak, sprawdziłem . @EqualsAndHashCode.Exclude na Rental w CompanyBranch pomaga i wtedy ładnie się zapisuje jak ta pętla jest też odpalona